### ISSUE TITLE
Cannot batch-convert S_HDMV/TEXTST subtitles

### ISSUE THREAD

**Issue by ipsi** [SubtitleEdit/subtitleedit#8039]
**TL;DR**: I can view and edit TextST (`S_HDMV/TEXTST`) subtitles via the UI, but *not* batch-extract them to SRT files via the CLI.

OS: Windows 11
SubtitleEdit: 4.0.4

I have a Bluray rip which contains both PGS and TextST subtitles. ~~Interestingly, it only seems to contain TextST subtitles for the audio commentary tracks, which is a bit surprising.~~ It's PGS for the movie audio track, and TextST for the commentary. I'd like to be able to dump out all of those subtitle tracks as they're not available any other way, but that doesn't seem to be easily possible.

Right now, if I try to do `SubtitleEdit.exe /convert '.\The Mummy_t00.mkv' SubRip`, it will assume I want to OCR the PGS subs, with no option to _not_ do that, that I can find. The workaround is to write a simple script to call SubtitleEdit once per track — annoying, but not a big deal.

Setting that aside, if I run `SubtitleEdit.exe /convert '.\The Mummy_t00.mkv' SubRip /track-number:76` instead, it _does_ produce an SRT file for that track, but it looks like this:

```
1
00:00:02,878 --> 00:00:06,048
�

2
00:00:06,089 --> 00:00:10,594
�
```

Repeated all the way down.

Which… isn't super-helpful? I know that SubtitleEdit supports these subtitles because I can load them via the GUI one at a time, but… There are four commentary tracks, a half dozen or so languages each, I'd like to extract _all_ of them, and doing them one by one will drive me bananas. The GUI “Batch convert” function also doesn't list the TextST tracks when loading up the file.

Is there something (possibly undocumented?) I can make them work via `/convert`? Or is there something missing from the converter? TextST subtitles don't appear to be very common, so I'm not terribly surprised that this is having some issues.

If this format can't be supported for batch conversions, I'd definitely appreciate an explicit error to that effect over a weird-looking output file!

**Comment by niksedk** [comment]
Thx for the info :)

Yes, `S_HDMV/TEXTST` for missing for cmd/batch convert.
Should now work in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.4/SubtitleEditBeta.zip
